### DISCUSSION TITLE
Is there any bot wide error handler for hybrid/slash commands similar to `Bot.on_command_error`?

### DISCUSSION THREAD

**Discussion by keqking** [Rapptz/discord.py#8384] [Q&A] [answered]
Hi,

currently i already have bot wide `on_command_error` to handle all errors from normal text commands so if a text command errors, I can send a template informative response to show to command author but is there any bot wide error listener for hybrid slash commands? When hybrid slash command errors, it just shows "This interaction failed" in red color, my goal is to ephemerally show some informative message to user if such listener exists. I searched in docs and found few:
* `cog_app_command_error` - would need to add/duplicate to every cog so not much useful to me
* `app_commands.Command.error` - per command deco, i think would need to add to every hybrid command
* `bot.tree.on_error` - seems its only called for `AppCommandError` and not for HybridCommandErrors? not sure if this will work for cog based hybrid command errors? 🤔 

None of these seem to fit my use case for a bot wide hybrid command error handler, would be immensely useful if there is one and i missed it in docs.
pls correct me if i'm wrong or I missed something. Thank you for reading.

**Comment by Rapptz** [comment]
`on_command_error` is already dispatched for hybrid commands as well.

**Reply by hirusha-adi** [reply]
what about slash commands?

**Reply by Rapptz** [reply]
No.

**Reply by hirusha-adi** [reply]
is there an alternative? for slash commands?

**Reply by PaulMarisOUMary** [reply]
Hi @hirusha-adi, you can create your own doing the following:
```py
# cogs/exception-handler.py

import discord
from discord.ext import commands

class ExceptionHandler(commands.Cog):
	def __init__(self, bot: commands.Bot):
		self.bot = bot

		bot.tree.error(coro = self.__dispatch_to_app_command_handler)

	async def __dispatch_to_app_command_handler(self, interaction: discord.Interaction, error: discord.app_commands.AppCommandError):
		self.bot.dispatch("app_command_error", interaction, error)

	@commands.Cog.listener("on_app_command_error")
	async def get_app_command_error(self, interaction: discord.Interaction, error: discord.app_commands.AppCommandError):
		pass # process exception here

async def setup(bot: commands.Bot):
    await bot.add_cog(ExceptionHandler(bot))
```

**Comment by hitblast** [comment, marked as answer]
You can make a separate cog for handling the exceptions inside your Discord bot.

> `on_command_error` is already dispatched for hybrid commands as well.

```python
# cogs/exception-handler.py

import discord
from discord.ext import commands

class ExceptionHandler(commands.Cog):
    def __init__(self, bot: commands.Bot) -> None:
        self.bot = bot

    @commands.Cog.listener()
    async def on_command_error(self, ctx: commands.Context, error) -> None:
        pass # do your things here
        
async def setup(bot: commands.Bot) -> None:
    await bot.add_cog(ExceptionHandler(bot))
```

**Comment by keqking** [comment]
thank you Danny and @hitblast that solved my question.

**Reply by hitblast** [reply]
Cheers!
